airflow-core/src/airflow/utils/state.py

```diff
@@ -44,12 +44,33 @@ def __str__(self) -> str:
         return self.value
 
 
+class SuccessTIState(str, Enum):
+    """States that a Task Instance can be in that indicate it has reached a terminal successful state."""
+
+    SUCCESS = "success"
+    SKIPPED = "skipped"  # A user can raise a AirflowSkipException from a task & it will be marked as skipped
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class FailureTIState(str, Enum):
+    """States that a Task Instance can be in that indicate it has reached a terminal failed state."""
+
+    FAILED = "failed"
+    REMOVED = "removed"
+
+    def __str__(self) -> str:
+        return self.value
+
+
 class IntermediateTIState(str, Enum):
     """States that a Task Instance can be in that indicate it is not yet in a terminal or running state."""
 
     SCHEDULED = "scheduled"
     QUEUED = "queued"
     RESTARTING = "restarting"
+    RUNNING = "running"
```

Comment:
`IntermediateTIState` is defined as States that a Task Instance can be in that indicate it is not yet in a terminal or running state.

We don't need RUNNING state, since the only way it can be in Execution Context is if task is marked running by the api-server